azurerm/resource_arm_autoscale_setting.go

```diff
@@ -0,0 +1,978 @@
+							MinItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"service_uri": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"properties": {
+										Type:     schema.TypeMap,
+										Optional: true,
+									},
+								},
+							},
+							Set: resourceAzureRmAutoscaleDefaultHash,
+						},
+					},
+				},
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Required: true,
+			},
+			"target_resource_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceArmAutoscaleSettingCreateOrUpdate(d *schema.ResourceData, meta interface{}) error {
+	armClient := meta.(*ArmClient)
+	asClient := armClient.autoscaleSettingsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name := d.Get("name").(string)
+	resourceGroupName := d.Get("resource_group_name").(string)
+	location := d.Get("location").(string)
+	enabled := d.Get("enabled").(bool)
+	targetResourceURI := d.Get("target_resource_id").(string)
+	tags := d.Get("tags").(map[string]interface{})
+	expandedTags := expandTags(tags)
+
+	profiles, err := expandAzureRmAutoscaleProfile(d)
+	if err != nil {
+		return err
+	}
+
+	notifications, err := expandAzureRmAutoscaleNotification(d)
+	if err != nil {
+		return err
+	}
+
+	autoscaleSetting := insights.AutoscaleSetting{
+		Name:              &name,
+		Enabled:           &enabled,
+		TargetResourceURI: &targetResourceURI,
+		Profiles:          &profiles,
+		Notifications:     &notifications,
+	}
+
+	parameters := insights.AutoscaleSettingResource{
+		Name:             &name,
+		Location:         &location,
+		Tags:             expandedTags,
+		AutoscaleSetting: &autoscaleSetting,
+	}
+
+	result, err := asClient.CreateOrUpdate(ctx, resourceGroupName, name, parameters)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(*result.ID)
+
+	return resourceArmAutoscaleSettingRead(d, meta)
+}
+
+func resourceArmAutoscaleSettingRead(d *schema.ResourceData, meta interface{}) error {
+	asClient := meta.(*ArmClient).autoscaleSettingsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	name := id.Path["autoscalesettings"]
+
+	if name == "" {
+		return fmt.Errorf("Cannot find resource name in Resource ID for Autoscale Setting")
+	}
+
+	result, err := asClient.Get(ctx, resGroup, name)
+	if err != nil {
+		if result.StatusCode == http.StatusNotFound {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request on Autoscale Setting %s: %+v", name, err)
+	}
+
+	if result.AutoscaleSetting == nil {
+		return fmt.Errorf("Unexpected result when reading Autoscale Setting %s: %#v", name, result)
+	}
+	autoscaleSetting := *result.AutoscaleSetting
+	d.Set("name", result.Name)
+	d.Set("resource_group_name", resGroup)
+	d.Set("location", result.Location)
```

Comment:
Set the location in a standard way like [the one in redis cache](https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/resource_arm_redis_cache.go#L427).